rotkehlchen/db/history_events.py

```diff
@@ -0,0 +1,225 @@
+                        amount,
+                        Asset(asset_name),
+                        Timestamp(int(high_precision_timestamp / KRAKEN_TS_MULTIPLIER)),
+                    ),
+                )
+            except DeserializationError as e:
+                log.error(
+                    f'Failed to read value from historic base entry {identifier} '
+                    f'with amount. {str(e)}',
+                )
+            except UnknownAsset as e:
+                log.error(
+                    f'Failed to read asset from historic base entry {identifier} '
+                    f'with asset identifier {asset_name}. {str(e)}',
+                )
+        return result
+
+    def get_entries_assets_history_events(
+        self,
+        query_filter: HistoryEventFilterQuery,
+    ) -> List[Asset]:
+        """Returns asset from base entry events using the desired filter"""
+        cursor = self.db.conn.cursor()
+        query, bindings = query_filter.prepare(with_pagination=False)
+        query = 'SELECT DISTINCT asset from history_events ' + query
+        result = cursor.execute(query, bindings)
+        assets = []
+        for asset_id in result:
+            try:
+                assets.append(Asset(asset_id[0]))
+            except (UnknownAsset, DeserializationError) as e:
+                self.db.msg_aggregator.add_error(
+                    f'Found asset {asset_id} in the base history events table and '
+                    f'is not in the assets database. {str(e)}',
+                )
+        return assets
+
+    def get_history_events_count(self, query_filter: HistoryEventFilterQuery) -> int:
+        """Returns how many of certain base entry events are in the database"""
+        cursor = self.db.conn.cursor()
+        query, bindings = query_filter.prepare(with_pagination=False)
+        query = 'SELECT COUNT(*) from history_events ' + query
+        result = cursor.execute(query, bindings)
+        return result.fetchone()[0]
+
+    def get_value_stats(
+        self,
+        query_filter: HistoryEventFilterQuery,
+    ) -> Tuple[FVal, List[Tuple[Asset, FVal]]]:
+        """Returns the sum of the USD value at the time of acquisition and the amount reveived
+        by asset"""
+        cursor = self.db.conn.cursor()
+        usd_value = FVal(0)
+        query_filters, bindings = query_filter.prepare(with_pagination=False)
+        try:
+            query = 'SELECT SUM(CAST(usd_value AS REAL)) FROM history_events ' + query_filters
+            result = cursor.execute(query, bindings)
+            usd_value = deserialize_fval(
+                value=result.fetchone()[0],
+                name='usd value in history events stats',
+                location='get_value_stats',
+            )
+        except (DeserializationError, sqlcipher.DatabaseError) as e:  # pylint: disable=no-member)
+            log.error(f'Didnt get correct valid usd_value for history_events query. {str(e)}')
```

Comment:
Hmmm what are you doing here? Log error and continue? This can't work. usd_value may have no value at this pointt. You need to reraise a DeserializationError probably and catch it upwards. Make the function raise DeserializationError.

Also probably then you won't need to catch DeserializatonError here